src/lib/Microsoft.Health.Fhir.Ingest/Service/MeasurementGroupFhirImportService.cs

```diff
@@ -139,17 +143,21 @@ private async Task ProcessMeasurementGroupEvents(IEnumerable<IEventMessage> even
                 try
                 {
                     IEnumerable<MeasurementGroup> groupedMeasurements;
+                    string partitionId;
 
                     try
                     {
-                        var partitionId = events.FirstOrDefault()?.PartitionId;
```

Comment:
The current code works, but wanted to get your thoughts on a future improvement.  Seems like we should consider creating a context or container object that wraps the events collection and would include additional information about the collection like the partition id to reduce code redundancy in the future.